### ISSUE TITLE
Game crash upon entering a specific world - "Ticking memory connection"

### ISSUE THREAD

**Issue by floral-qua-floral** [Hunternif/AntiqueAtlas#314]
**Describe the bug**
Last time I played singleplayer in my modpack, I crafted an Atlas and added a few waypoints. Today it appears I'm unable to enter my world anymore, as the game will close to the launcher immediately as soon as the game reaches "Joining world". The crash report seems to indicate this is related to AA's global marker data handler, and removing the mod allows me to enter my world without issue. Deleting the 7 atlas-related files from my save's \data\ folder (after making a backup) does not allow me to enter the world.

**To Reproduce**
Regrettably, I can't seem to get the issue to occur in a new world - I'm able to create a new world, spawn an Atlas, and add some markers, all without crashing. The crash only seems to occur in my Singleplayer survival save (although it is 100% consistent - it occurs immediately upon joining the world). I have made a backup of the world and loaded it with most mods removed and confirmed that the crash does still occur.

**Screenshots**
Launcher displays this popup after crash:
![image](https://user-images.githubusercontent.com/49110090/106785812-d34b2700-661b-11eb-984f-8091796acce2.png)

**Forge version information (please complete the following information):**
 - Minecraft: 1.16.5
 - Forge: 36.0.1 (Forge version packaged with Vivecraft - however this testing was all done _without_ Vivecraft, on the normal Forge version). Also tested with Forge 36.0.14, the latest 1.16.5 build, which doesn't seem to change anything - crashes still occur with very similar reports.
 - AntiqueAtlas: 5.4.3-forge

**Additional files**
Crash report when world is loaded with the full modpack:
[crash-2021-02-03_11.54.39-server.txt](https://github.com/AntiqueAtlasTeam/AntiqueAtlas/files/5920376/crash-2021-02-03_11.54.39-server.txt)

Crash report when world is loaded with only worldgen-related mods:
[crash-2021-02-03_12.20.26-server.txt](https://github.com/AntiqueAtlasTeam/AntiqueAtlas/files/5920377/crash-2021-02-03_12.20.26-server.txt)

Client crash reports also seem to generate sometimes with a "Lock is no longer valid" error, but I'm having a hard time pinpointing what causes them to show up as opposed to just server crash reports. This crash report occurred when attempting to load the world with the full modpack, immediately after updating to Forge 36.0.14:
[crash-2021-02-03_12.53.18-client.txt](https://github.com/AntiqueAtlasTeam/AntiqueAtlas/files/5920494/crash-2021-02-03_12.53.18-client.txt)

The `data` folder from the world's save seems to have some Atlas-related files in there, so I felt it might be worth including it as well in case they could shed some light on what's going on:
[data.zip](https://github.com/AntiqueAtlasTeam/AntiqueAtlas/files/5920407/data.zip)

After deleting all these files from \data\ and attempting to load the world again, only two of them seem to have been created fresh: aAtlasGlobalMarkers.dat and aAtlasTiles. The other 5 atlas files remain missing - here are the two that did generate:
[data2.zip](https://github.com/AntiqueAtlasTeam/AntiqueAtlas/files/5920514/data2.zip)

If there are any other files or information that I could share that could be useful, let me know.

**Comment by tyra314** [comment]
Thank you for the *very* good bug report. I wish every bug report would be like this.

This has the same issue as #309 and will be fixed with #310, which has some other issues blocking it though.

**Comment by AlyceOsbourne** [comment]
Just leaving my applause at your thoroughness <3

**Comment by DestinyPlayer** [comment]
https://pastebin.com/XU2wEPkn - Crash Report
https://prnt.sc/12n01qz - Modpack
I have a similar issue on Forge, except mine only happens when I have BSL Shaders OFF. When I turn them ON, it lets me enter the world.

**Comment by P4YN-Ki** [comment]
That is an extremely awesome report. I'm having the same issue as OP except I can pin point the "Lock is no longer valid" issue. That seem to happen with AA and Optifine. If I remove one or the other, then that report no longer appears and I only get the "Ticking Memory Connection" as of which Minecraft continues to use all 32GB of my RAM and then eventually crashes upon loading into the world. The exact line of the crash log that points to AA reads as follows

"at hunternif.mc.impl.atlas.marker.GlobalMarkersDataHandler.onPlayerLogin(GlobalMarkersDataHandler.java:59) ~[antiqueatlas:?] {re:classloading}"

I'm unsure if this will help any, but I can provide more detailed information if it's needed.

**Comment by KirbyBrasse** [comment]
> Thank you for the _very_ good bug report. I wish every bug report would be like this.
> 
> This has the same issue as #309 and will be fixed with #310, which has some other issues blocking it though.

I read #310 but doesn't understand how to fix the problem with what is written in 310

**Comment by Wedge1453** [comment]
so it isn't fixed yet?

**Comment by Teoxex** [comment]
> so it isn't fixed yet?

No, sadly, we need to wait

**Comment by MozarteanChaos** [comment]
i'm having a similar issue, so i've temporarily uninstalled the mod.
i'm not sure if a log would be useful since there's already been a couple posted (although i can post it if it'd help at all), but i _can_ say that crafting or otherwise obtaining an atlas isn't necessary for the glitch to occur.

**Comment by mydah42** [comment]
I'm really glad this issue is listed is here. I was having the same problem and didn't realize it was the Antique Atlas causing the error. 

I am new to modding Minecraft. I got setup with the modpack "PwrDown's RPG". It's great! After playing it for a couple weeks I saw the Antique Atlas mod and I realized it would be a perfect addition! I added the Atlas and it was FANSTSTIC for a week or so. Then my game started crashing with Exit Code: 0.  I did not immediately make the connection with the Atlas because it had been working fine a while.

I got four of the crash logs and tried, and failed, to understand it. After a lot of Googleing I came across this issue post from Floral-qua-floral. I removed Antique Atlas and, much to my disappointment, my game started working normally right away. I played all day today without problems. Anique Atlas is FANTASTIC and I wish I could keep using it!

Since I'm new to Minecraft modding and I didn't know what I was doing, I made regular backups of the world, both before and after I installed the Atlas. I can share them if it would help.

[crash-2021-07-14_22.18.06-server.txt](https://github.com/AntiqueAtlasTeam/AntiqueAtlas/files/6827422/crash-2021-07-14_22.18.06-server.txt)
[crash-2021-07-14_22.27.18-server.txt](https://github.com/AntiqueAtlasTeam/AntiqueAtlas/files/6827423/crash-2021-07-14_22.27.18-server.txt)
[crash-2021-07-14_22.38.36-server.txt](https://github.com/AntiqueAtlasTeam/AntiqueAtlas/files/6827424/crash-2021-07-14_22.38.36-server.txt)
[crash-2021-07-14_22.59.41-server.txt](https://github.com/AntiqueAtlasTeam/AntiqueAtlas/files/6827426/crash-2021-07-14_22.59.41-server.txt)

**Comment by rabbivole** [comment]
Hello! I think I might be having _exactly_ the issue the OP had, so I figured I'd throw my crashlog and information onto the pile on the offchance it might be useful.

[crash-2021-07-19_20.26.34-server.txt](https://github.com/AntiqueAtlasTeam/AntiqueAtlas/files/6845301/crash-2021-07-19_20.26.34-server.txt)

Singleplayer 1.16.5, using the official Mojang launcher, Forge 36.1.0, AntiqueAtlas 5.4.4. I'm not using a modpack; I got a wild hair up my ass to play some minecraft and dropped in some mods I wanted by hand. I mention this because I had to get creative with older versions on a few things to get everything to boot and haven't done any of the extra work/quality control that some modpacks do to make sure everything plays nice together. I can't guarantee I haven't set up some weird and unusual conditions and only kinda sorta know what I'm doing. 

I generated the world, played, and quit/restarted several times without problems. After about 12 hours of total playtime and a fair amount of exploring, I quit (in a village with my Atlas in one of my hotbar slots). Upon attempting to load the world again, I had the exact same crash as the original poster 100% of the time: world loads, the UI starts to appear and the sky loads in, then the game closes silently with the "Exit code: 0" popup in the official launcher. Based on vague memories of what 'ticking memory connection' can mean, I tried some stuff that didn't help: rebooted my computer, kept the window small while loading in, and set my view distance to 2 chunks. Googling led me here, I removed the Antique Atlas jar from /mods/, and I immediately loaded in fine. 

Bummer, because Antique Atlas is really delightful and I'd love to keep using it. Good luck!
